plugins/modules/foreman_installation_medium.py

```diff
@@ -124,12 +130,15 @@ def main():
 
     module.connect()
     name = entity_dict['name']
+    entity = None
```

Comment:
why is this? I think we want to be extra sure, that `entity` as a variable is not even defined, in case of `affects_multiple`.

Reply:
If this variable is not defined, I've an `UnboundLocalError: local variable 'entity' referenced before assignment` with python 3 because I check that entity is not None before applying the updated name.

Reply:
I don't understand why it should only be problematic in py3? Also it should not be needed if you move that transformation in one of the `not affects_multiple` blocks.

Reply:
Rewritten condition so that's not needed anymore.